components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -786,33 +659,46 @@ public void testChown(boolean isSudoing, boolean permChown,
+            assertOwnedBy(annotOriginalFileAnnotationTagAndLinks, normalUser);
+        } else if (chownPassingWhenNotSudoing && annotationsChownExpectSuccess) {
             /* chowning the image NOT being sudoed,
              * should pass only in case you have Chown
              * privilege, captured in "chownPassingWhenNotSudoing" boolean.
              * Also check that the value of canChown boolean matches chownPassingWhenNotSudoing
              * boolean in such case.*/
-            Assert.assertEquals(getCurrentPermissions(image).canChown(), chownPassingWhenNotSudoing);
+            Assert.assertEquals(getCurrentPermissions(image).canChown(), true);
             doChange(client, factory, Requests.chown().target(image).toUser(anotherUser.userId).build(), chownPassingWhenNotSudoing);
-            if (chownPassingWhenNotSudoing) {
-                assertOwnedBy(image, anotherUser);
-                assertOwnedBy((new OriginalFileI(remoteFileId, false)), anotherUser);
-            } else {
-                assertOwnedBy(image, normalUser);
-                assertOwnedBy((new OriginalFileI(remoteFileId, false)), normalUser);
-            }
-            /* in any case, the image must be in the right group */
-            long imageGroupId = ((RLong) iQuery.projection(
-                    "SELECT details.group.id FROM Image i WHERE i.id = :id",
-                    new ParametersI().addId(image.getId())).get(0).get(0)).getValue();
-            Assert.assertEquals(imageGroupId, normalUser.groupId);
+            assertOwnedBy(image, anotherUser);
+            assertOwnedBy(originalFile, anotherUser);
+            /* for the annotations on the image and their links,
+             * the chown of the image will only chown them if also
+             * groupPermissions are private or read-only (captured in boolean
+             * annotationsChownExpectSuccess) */
+            assertOwnedBy(annotOriginalFileAnnotationTagAndLinks, anotherUser);
+        } else if (chownPassingWhenNotSudoing && !annotationsChownExpectSuccess){
+            Assert.assertEquals(getCurrentPermissions(image).canChown(), true);
+            doChange(client, factory, Requests.chown().target(image).toUser(anotherUser.userId).build(), chownPassingWhenNotSudoing);
+            assertOwnedBy(image, anotherUser);
+            assertOwnedBy(originalFile, anotherUser);
+            /* in read-annotate and read-write groups the
+             * annotations will remain attached to the image
+             * of anotherUser, but still belongs to normalUser
+             */
+            assertOwnedBy(annotOriginalFileAnnotationTagAndLinks, normalUser);
+        } else {
+            /* the chown will fail, as the chownPassingWhenNotSudoing is false,
+             * and the light admin is not sudoing in this case. All objects still
+             * belong to normalUser. */
+            Assert.assertEquals(getCurrentPermissions(image).canChown(), false);
```

Comment:
Fixed in https://github.com/pwalczysko/openmicroscopy/commit/04efe7400c43be1073ef47ee97356683a906ff0d#diff-d91232ef1e245a87e311c4a79a2a5f77L692